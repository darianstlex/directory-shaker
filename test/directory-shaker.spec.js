const path = require('path');
const getTreeObj = require('../src/directory-shaker');

const store = path.resolve(__dirname, 'store');

describe('Directory Shaker', () => {
  it('should return correct tree by default', () => {
    expect(getTreeObj(store)).toMatchSnapshot();
  });

  it('should return correct tree if not recursive', () => {
    expect(getTreeObj(store, { recursive: false })).toMatchSnapshot();
  });

  it('should return correct tree if index not spreaded', () => {
    expect(getTreeObj(store, {
      recursive: false,
      spreadIndex: false,
    })).toMatchSnapshot();
  });

  it('should return correct tree on provided pattern', () => {
    expect(getTreeObj(store, { pattern: /^.+\.mjs$/ })).toMatchSnapshot();
  });

  it('should return correct tree on filter', () => {
    expect(getTreeObj(store, { filter: ({ name }) => name })).toMatchSnapshot();
  });

  it('should return correct tree and extra data', () => {
    expect(getTreeObj(store, {
      filter: ({ name }) => name
    }, {
      name: 'paths',
      key: 'name',
      map: ({ path, name, page }) => ({ path, name, page }),
    })).toMatchSnapshot();
  });

  it('should return correct tree and more extra data', () => {
    expect(getTreeObj(store, {
      filter: ({ name }) => name
    }, {
      name: 'paths',
      key: 'name',
      map: ({ path, name, page }) => ({ path, name, page }),
    }, {
      name: 'pages',
      key: 'name',
      map: ({ path }) => path,
    })).toMatchSnapshot();
  });
});
