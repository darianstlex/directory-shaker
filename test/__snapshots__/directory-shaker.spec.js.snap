// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory Shaker should return correct tree and extra data 1`] = `
Object {
  "paths": Object {
    "aboutPage": Object {
      "name": "aboutPage",
      "page": "About Us",
      "path": "/about",
    },
    "batteryAA": Object {
      "name": "batteryAA",
      "page": "Battery AA",
      "path": "/products/battery",
    },
    "droneSuper": Object {
      "name": "droneSuper",
      "page": "Drone Super",
      "path": "/products/drone",
    },
    "storeRoot": Object {
      "name": "storeRoot",
      "page": "Root Page",
      "path": "/",
    },
  },
  "tree": Object {
    "about": Object {
      "name": "aboutPage",
      "page": "About Us",
      "path": "/about",
    },
    "name": "storeRoot",
    "page": "Root Page",
    "path": "/",
    "products": Object {
      "battery": Object {
        "header_color": "some color",
        "header_name": "some name",
        "header_text": "some text",
        "name": "batteryAA",
        "page": "Battery AA",
        "path": "/products/battery",
      },
      "drone": Object {
        "getPrice": [Function],
        "header_color": "some color",
        "header_name": "some name",
        "header_text": "some text",
        "name": "droneSuper",
        "page": "Drone Super",
        "path": "/products/drone",
      },
    },
  },
}
`;

exports[`Directory Shaker should return correct tree and more extra data 1`] = `
Object {
  "pages": Object {
    "aboutPage": "/about",
    "batteryAA": "/products/battery",
    "droneSuper": "/products/drone",
    "storeRoot": "/",
  },
  "paths": Object {
    "aboutPage": Object {
      "name": "aboutPage",
      "page": "About Us",
      "path": "/about",
    },
    "batteryAA": Object {
      "name": "batteryAA",
      "page": "Battery AA",
      "path": "/products/battery",
    },
    "droneSuper": Object {
      "name": "droneSuper",
      "page": "Drone Super",
      "path": "/products/drone",
    },
    "storeRoot": Object {
      "name": "storeRoot",
      "page": "Root Page",
      "path": "/",
    },
  },
  "tree": Object {
    "about": Object {
      "name": "aboutPage",
      "page": "About Us",
      "path": "/about",
    },
    "name": "storeRoot",
    "page": "Root Page",
    "path": "/",
    "products": Object {
      "battery": Object {
        "header_color": "some color",
        "header_name": "some name",
        "header_text": "some text",
        "name": "batteryAA",
        "page": "Battery AA",
        "path": "/products/battery",
      },
      "drone": Object {
        "getPrice": [Function],
        "header_color": "some color",
        "header_name": "some name",
        "header_text": "some text",
        "name": "droneSuper",
        "page": "Drone Super",
        "path": "/products/drone",
      },
    },
  },
}
`;

exports[`Directory Shaker should return correct tree by default 1`] = `
Object {
  "tree": Object {
    "about": Object {
      "name": "aboutPage",
      "page": "About Us",
      "path": "/about",
    },
    "name": "storeRoot",
    "page": "Root Page",
    "path": "/",
    "products": Object {
      "battery": Object {
        "header_color": "some color",
        "header_name": "some name",
        "header_text": "some text",
        "name": "batteryAA",
        "page": "Battery AA",
        "path": "/products/battery",
      },
      "drone": Object {
        "getPrice": [Function],
        "header_color": "some color",
        "header_name": "some name",
        "header_text": "some text",
        "name": "droneSuper",
        "page": "Drone Super",
        "path": "/products/drone",
      },
      "header": Object {
        "header_color": "some color",
        "header_name": "some name",
        "header_text": "some text",
      },
    },
  },
}
`;

exports[`Directory Shaker should return correct tree if index not spreaded 1`] = `
Object {
  "tree": Object {
    "about": Object {
      "name": "aboutPage",
      "page": "About Us",
      "path": "/about",
    },
    "index": Object {
      "name": "storeRoot",
      "page": "Root Page",
      "path": "/",
    },
  },
}
`;

exports[`Directory Shaker should return correct tree if not recursive 1`] = `
Object {
  "tree": Object {
    "about": Object {
      "name": "aboutPage",
      "page": "About Us",
      "path": "/about",
    },
    "name": "storeRoot",
    "page": "Root Page",
    "path": "/",
  },
}
`;

exports[`Directory Shaker should return correct tree on filter 1`] = `
Object {
  "tree": Object {
    "about": Object {
      "name": "aboutPage",
      "page": "About Us",
      "path": "/about",
    },
    "name": "storeRoot",
    "page": "Root Page",
    "path": "/",
    "products": Object {
      "battery": Object {
        "header_color": "some color",
        "header_name": "some name",
        "header_text": "some text",
        "name": "batteryAA",
        "page": "Battery AA",
        "path": "/products/battery",
      },
      "drone": Object {
        "getPrice": [Function],
        "header_color": "some color",
        "header_name": "some name",
        "header_text": "some text",
        "name": "droneSuper",
        "page": "Drone Super",
        "path": "/products/drone",
      },
    },
  },
}
`;

exports[`Directory Shaker should return correct tree on provided pattern 1`] = `
Object {
  "tree": Object {
    "products": Object {
      "new": Object {
        "name": "newProduct",
        "page": "New Product",
        "path": "products/new",
      },
    },
  },
}
`;
